File: app/src/main/java/com/example/kotlin_notes/NoteAdapter.java
package com.example.kotlin_notes;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.bitmap.CenterCrop;
import com.bumptech.glide.load.resource.bitmap.RoundedCorners;

import java.util.ArrayList;
import java.util.List;

public class NoteAdapter extends RecyclerView.Adapter<NoteAdapter.viewHolder> {

    private List<Note> list = new ArrayList<>();

    public void setList(List<Note> list) {
        //this.list= list;
        this.list.clear();
        this.list.addAll(list);
        notifyDataSetChanged();
    }

    public void addNote(Note note) {
        list.add(note);
        notifyDataSetChanged();
    }


    public Note getItem(int pos) {

        return list.get(pos);
    }


    @NonNull
    @Override
    public viewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_note, parent, false);
        return new viewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull viewHolder holder, int position) {
        holder.bind(position);

    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class viewHolder extends RecyclerView.ViewHolder {
        private ImageView imageView;
        private TextView title;
        private TextView desc;
        //new
        private ImageView delete;
        private ImageView pen;


        public viewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.item_image);
            title = itemView.findViewById(R.id.item_title);
            desc = itemView.findViewById(R.id.item_desc);
            delete = itemView.findViewById(R.id.item_delete);
            pen = itemView.findViewById(R.id.item_pen);
        }



        void bind(int position) {
            title.setText(list.get(position).getTitle());
            desc.setText(list.get(position).getDesc());



            Glide
                    .with(itemView)
                    .load(list.get(position)
                            .getImage())
                    .transform(new CenterCrop(), new RoundedCorners(25))
                    .into(imageView);

            delete.setOnClickListener(v -> {
                list.remove(position);
                notifyItemRemoved(position);
            });

        }
    }
}

File: app/src/main/java/com/example/kotlin_notes/NoteAdapter.kt
package com.example.kotlin_notes

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.resource.bitmap.CenterCrop
import com.bumptech.glide.load.resource.bitmap.RoundedCorners

open class NoteAdapter(private  val listener: IOnItem) :
    RecyclerView.Adapter<NoteAdapter.ViewHolder>() {

    val list : MutableList<NoteModel> = ArrayList()



    fun setList(list: MutableList<NoteModel>) {
        this.list .clear()
        this.list.addAll(list)
        notifyDataSetChanged()

    }


    fun addNote(note: NoteModel){
        list.add(note)
        notifyDataSetChanged()
    }


    fun delete(pos: Int) {
        list.removeAt(pos)
        notifyItemRemoved(pos)
    }

    fun getItem(pos:Int):NoteModel{
        return list[pos]
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
       holder.bind(position)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =LayoutInflater.from(parent.context).inflate(R.layout.item_note,parent,false)
        return ViewHolder(view)
    }



    inner  class ViewHolder (itemView : View):RecyclerView.ViewHolder(itemView){
        private  val imageView: ImageView =itemView.findViewById(R.id.item_image)
        private  val title : TextView = itemView.findViewById(R.id.item_title)
        private  val  desc : TextView = itemView.findViewById(R.id.item_desc)
        private  val  delete :ImageView = itemView.findViewById(R.id.item_delete)
        private  val  pen : ImageView = itemView.findViewById(R.id.item_pen)
        private val share : ImageView = itemView.findViewById(R.id.share)


        fun bind(pos: Int){
            title.text = list[position].title
            desc.text= list[position].desc
            Glide.with(itemView)
                .load(list[position].image)
                .transform(CenterCrop(), RoundedCorners(25))
                .into(imageView)

            delete.setOnClickListener{
                listener.delete(adapterPosition)
            }

            pen.setOnClickListener {
                listener.edit(adapterPosition, list[position]) }


            share.setOnClickListener{
                listener.share(adapterPosition)
            }
        }




    }

    interface IOnItem{
        fun  delete(pos:Int)
        fun edit(pos:Int, note: NoteModel)
        fun share (pos:Int)
    }

}